### PR TITLE
DO NOT MERGE F8 always runs in active terminal

### LINKED ISSUE #1607: I want F8 to run selected text in the active terminal, not just in the "Integrated PowerShell Console"
I would like F8 to run a selected text in any active terminal, not just in the "Integrated PowerShell Console".

I use more and more PowerShell Core in WSL terminal in VSCode and it would be nice to use F8 to run selected text in that terminal too.

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
## PR Summary

fixes #1607

Note: I decided against having this behavior to be configurable. The thought was that most people will only use the PowerShell Integrated Console.

In #1607 I brought up worry about people getting confused that their intellisense cache was not updating if they use pwsh instead of PSIC... I think perhaps this is a worry that does not exist and if it does and they open an issue, we can educate them.

Thoughts?

~NOTE this relies on an engine update to 1.29 which is the current release. We probably don't want to merge this in until 1.30 is released~
1.30 has been released

~NOTE I seem to have found a potential bug (Microsoft/vscode#64390) in VSCode with this change... can't merge this in until it's sorted~

The bug has been fixed. We can probably merge this in when 1.31.0 drops.

## PR Checklist

Note: Tick the boxes below that apply to this pull request by putting an `x` between the square brackets.
Please mark anything not applicable to this PR `NA`.

- [x] PR has a meaningful title
- [x] Summarized changes
- [ ] PR has tests - N/A
- [x] This PR is ready to merge and is not work in progress
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
cc @alexandair

File: src/features/Console.ts [hunk 1]
```diff
@@ -210,6 +211,17 @@ export class ConsoleFeature implements IFeature {
                     return;
                 }
 
+                if (vscode.window.activeTerminal.name !== "PowerShell Integrated Console") {
+                    this.log.write("PSIC is not active terminal. Running in active terminal using 'runSelectedText'");
+                    await vscode.commands.executeCommand("workbench.action.terminal.runSelectedText");
+
+                    // we need to honor the focusConsoleOnExecute setting here too.
+                    vscode.window.activeTerminal.show(!Settings.load().integratedConsole.focusConsoleOnExecute);
```
Comment (rkeithhill) [inline src/features/Console.ts]: So by calling `show()` it doesn't put focus on that console?

Comment (TylerLeonhardt) [reply src/features/Console.ts]: The opposite. The parameter name is `preserveFocus` which, if `true` will keep the focus where it is. if `false` it will NOT preserver the focus and put focus on the terminal. That's why I needed to reverse the boolean.

Comment (rjmholt) [reply src/features/Console.ts]: Is there a way to show the name of that parameter here? As it is, it's a bit hard to follow

Comment (TylerLeonhardt) [reply src/features/Console.ts]: sadly there are [no named params in typescript](https://github.com/Microsoft/TypeScript/issues/467)... but I added a longer comment to explain why we need to inverse it.

**Comment by alexandair** [comment]
@TylerLeonhardt Thank you so much. It will make it much easier to use various shells. Azure Cloud Shell, I'm looking at you. :)

It will work with multiple lines when a backtick is used as a continuation character, right?

**Comment by TylerLeonhardt** [comment]
I seem to have found a potential bug in VSCode with this change... can't merge this in until it's sorted:

https://github.com/Microsoft/vscode/issues/64390

**Comment by alexandair** [comment]
@TylerLeonhardt Excellent.
